documentation/manual/src/main/asciidoc/modules/mapping.asciidoc

```diff
@@ -120,3 +120,27 @@ store 1 million tuples in the association key.
 For example the Infinispan Embedded dialect suffers from this limitation as it's
 treated as a pure key/value store, while the Infinispan Remote dialect does not
 as it is more similar to a document store.
+
+==== Elements order in associations
+
+Hibernate OGM, by default, does not guarantees that elements in an association will
+be retrieved in the same order each time you read the association.
+
+If the order is important, you can enforce it using the following annotations:
+
+* `@javax.persistence.OrderColumn`: the collection uses a dedicated order column
+  in the collection link table
+
+* `@javax.persistence.OrderBy`: the collection is ordered upon retrieval using
+  a child entity property
+
+You can find more details and examples in the
+http://docs.jboss.org/hibernate/orm/5.3/userguide/html_single/Hibernate_User_Guide.html#collections-list[Hibernate ORM documentation].
+
+[WARNING]
+====
+At the moment, Hibernate OGM doesn't support associations of embeddables with duplicates.
```

Comment:
> According to the tests we don't guarantee for duplicates keeping even if we have a classic OneToMany. Do we? 

Yes, we do. The duplicates are only a problem with embeddables because they don't have a column that can be used as an id. That's why with the `@OrderColumn` annotation everything works. Because the column for the order acts as an id for the element.

For associations with entities we always a column or an id and therefore the problem doesn't occur. The test we added prove otherwise.

> without the annotationswe leave to the datastore the burden of handle it. Could we keep this idea?

This idea is wrong or at the very least misleading. Even if the datastore supports duplicates (and I think all our datastores do) when Hibernate OGM reads the association it will lose the duplicates and only keep one copy. This happens because, in the case where we don't have a column that acts as id, the RowKey generated for the duplicates will be the same and override each other.

Reply:
Ah... You are actually right.

> For associations with entities we always have a column or an id and therefore the problem doesn't occur. The test we added prove otherwise.

The rowkey will be the same in for these cases. I will update the tests